common/meta/api/src/schema_api_impl.rs

```diff
@@ -192,6 +194,80 @@ impl<KV: KVApi> SchemaApi for KV {
         }
     }
 
+    async fn rename_database(
+        &self,
+        req: RenameDatabaseReq,
+    ) -> Result<RenameDatabaseReply, MetaError> {
+        let tenant_dbname = &req.name_ident;
+        let tenant_newdbname = DatabaseNameIdent {
+            tenant: tenant_dbname.tenant.clone(),
+            db_name: req.new_db_name.clone(),
+        };
+
+        loop {
+            // get old db, not exists return err
+            let res = get_db_or_err(
+                self,
+                tenant_dbname,
+                format!("rename_database: {}", &tenant_dbname),
+            )
+            .await;
+
+            let (old_db_id_seq, old_db_id, _, _) = match res {
+                Ok(x) => x,
+                Err(e) => {
+                    if let MetaError::AppError(AppError::UnknownDatabase(_)) = e {
+                        if req.if_exists {
+                            return Ok(RenameDatabaseReply {});
+                        }
+                    }
+                    return Err(e);
+                }
+            };
```

Comment:
Since `db_meta` is never used, this part can be simplified to `let res = get_id_value(self, tenant_db_name...) ...`, which will save one RPC to metasrv.

But it's not a big deal and can be done in next PR.